lib/Domain/Prototype/Type.php

```diff
@@ -14,14 +16,29 @@ final class Type extends Prototype
      */
     private $none = false;
 
-    public function __construct(string $type = null)
+    /**
+     * @var bool
+     */
+    private $nullable = false;
+
+    private function __construct(string $type = null, bool $nullable = false)
     {
         $this->type = $type;
+        $this->nullable = $nullable;
     }
 
-    public static function fromString(string $string): Type
+    public static function fromString(string $string, bool $nullableAllowed = false): Type
```

Comment:
Do we need the `nullableAllowed`?

Not sure it's the Type classes responsiblity to check this (and would be good to remove the extra responsiblity from this static constuctor).

Or maybe there's a good reason to keep it?

Reply:
Good question : I thought maybe some use cases do not allow nullable types but maybe I was wrong. I can remove it completely and allow nullable types in any case.

Reply:
I guess a nullable type is technically a collection of two types f.e. `string|null`. So technically it should be modelled as `Types::fromArray([Type::fromString('foobar'), Type::null()])` (or `Types::fromString('?foobar')`.

But for the purposes of this library (at least now) I think having nullable as a property of the type is fine :)

Reply:
Interesting. But maybe it would make things more complicated in a way, because a function like `public function getSomething(): \Foo\Bar|string|null;` in an interface and a class is not allowed in PHP. Gonna change the code to remove nullableAllowed :)